rskj-core/src/test/java/co/rsk/peg/BridgeTest.java

```diff
@@ -81,6 +81,65 @@ public void getActivePowpegRedeemScript_after_RSKIP293_activation() throws VMExc
         assertArrayEquals(constants.bridgeConstants.getGenesisFederation().getRedeemScript().getProgram(), result);
     }
 
+    @Test
+    public void getPegoutCreationRskTxHashByBtcTxHash_before_RSKIP298_activation() throws VMException {
+        doReturn(false).when(activationConfig).isActive(eq(RSKIP298), anyLong());
+
+        BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
+        Bridge bridge = getBridgeInstance(bridgeSupportMock, activationConfig);
+
+        Sha256Hash btcTxHash = PegTestUtils.createHash(30);
+
+        byte[] data = BridgeMethods.GET_PEGOUT_CREATION_RSK_TX_HASH_BY_BTC_TX_HASH.getFunction().encode(
+            btcTxHash.getBytes()
+        );
+
+        assertNull(bridge.execute(data));
+    }
+
+    @Test
+    public void getPegoutCreationRskTxHashByBtcTxHash_after_RSKIP298_activation() throws VMException {
+        doReturn(true).when(activationConfig).isActive(eq(RSKIP298), anyLong());
+
+        Sha256Hash btcTxHash = PegTestUtils.createHash(13);
+        Keccak256 rskTxHash = PegTestUtils.createHash3(7);
+
+        BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
+        when(bridgeSupportMock.getPegoutCreationRskTxHashByBtcTxHash(any())).thenReturn(
+            Optional.of(rskTxHash)
+        );
+
+        Bridge bridge = getBridgeInstance(bridgeSupportMock, activationConfig);
+
+        byte[] data = BridgeMethods.GET_PEGOUT_CREATION_RSK_TX_HASH_BY_BTC_TX_HASH.getFunction().encode(
+            btcTxHash.getBytes()
+        );
+        byte[] result = (byte[]) BridgeMethods.GET_PEGOUT_CREATION_RSK_TX_HASH_BY_BTC_TX_HASH.getFunction().decodeResult(bridge.execute(data))[0];
+
+        assertArrayEquals(rskTxHash.getBytes(), result);
+    }
+
+    @Test
+    public void getPegoutCreationRskTxHashByBtcTxHash_not_found_after_RSKIP298_activation() throws VMException {
```

Comment:
If you end up implementing the change I suggested handling the response from the provider in BridgeSupport, then this test would not be needed anymore here